src/stream/src/executor/sync_kv_log_store.rs

```diff
@@ -404,12 +471,26 @@ impl<S: StateStore> SyncedKvLogStoreExecutor<S> {
                                     epoch,
                                 );
                                 flush_info.report(&self.metrics);
-                                write_future =
+                                write_future_state =
                                     WriteFuture::receive_from_upstream(stream, write_state);
                             }
                         }
                     }
                     Either::Right(result) => {
+                        if !clean_state
+                            && matches!(read_future_state, ReadFuture::Idle)
+                            && buffer.no_flushed_items()
+                        {
+                            clean_state = true;
+
+                            // Let write future resume immediately
+                            if let WriteFuture::Paused { sleep_future, .. } =
+                                &mut write_future_state
+                            {
+                                assert!(buffer.current_size < self.max_buffer_size);
+                                *sleep_future = Box::pin(sleep(Duration::ZERO));
```

Comment:
Not sure about whether it's illegal to sleep with zero duration. May need to add a ut to simply call `sleep(Duration::ZERO)`. Or can we just make `sleep_future` Option, and when we pull the future, if we see it's None, we can return ready immediately.

Reply:
Option sounds cleaner. Let's go with it.

Reply:
Fixed